### PR TITLE
refactor(memline): convert function comments to doxygen format

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

**Comment by zeertzjq** [comment]
Wait a minute, there is something wrong with `ml_close_all()`.

**Comment by zeertzjq** [comment]
#17745

File: src/nvim/memline.c [hunk 1]
```diff
@@ -1616,8 +1596,8 @@ static time_t swapfile_info(char_u *fname)
   return x;
 }
 
-/// Returns TRUE if the swap file looks OK and there are no changes, thus it
-/// can be safely deleted.
+/// @return  TRUE if the swap file looks OK and there are no changes, thus it
+///          can be safely deleted.
 static time_t swapfile_unchanged(char *fname)
```
Comment (zeertzjq) [inline src/nvim/memline.c:1601]: I just noticed this function had a wrong return type (<https://github.com/vim/vim/pull/10000>). That's may I marked this as draft again.

Comment (dundargoc) [reply src/nvim/memline.c:1601]: Aha, thought it was a "you can undraft this" ping :). Redrafting again then, let me know when this is good to go.

File: src/nvim/memline.c [hunk 2]
```diff
@@ -1616,9 +1596,9 @@ static time_t swapfile_info(char_u *fname)
   return x;
 }
 
-/// @return  true if the swap file looks OK and there are no changes, thus it
+/// @return  TRUE if the swap file looks OK and there are no changes, thus it
 ///          can be safely deleted.
-static bool swapfile_unchanged(char *fname)
+static time_t swapfile_unchanged(char *fname)
```
Comment (zeertzjq) [inline src/nvim/memline.c:1601]: This no longer needs change.

**Pull request merged**